src/middlewared/middlewared/plugins/account_/2fa.py

```diff
@@ -29,7 +29,9 @@ async def provisioning_uri(self, username):
         """
         user = await self.translate_username(username)
         user_twofactor_config = await self.middleware.call(
-            'auth.twofactor.get_user_config', user['id' if user['local'] else 'sid'], user['local'],
+            'auth.twofactor.get_user_config',
+            user['id' if user['local'] else 'sid'],
```

Comment:
I don't like these types of comprehensions when we're checking the same variable twice. It's also kind of messy code. What do you think about doing something like:
```
user_id, local_user = 'id', 'DB_ID'
if not user['local']:
    user_id, local_user = 'sid', 'SID'

user_twofactor_config = await self.middleware.call(
    'auth.twofactor.get_user_config', user_id, local_user
)

Reply:
I'm actually testing backing out the 2fa changes right now.